Take two of preliminary packagification steps

javh opened the pull request:
Initial packagification steps, take two.  Changes are largely the same things done in pull request #17, with some modifications (like putting all scripts into `inst/exec` instead of language specific subfolders).

The examples in the quickstart seem to work fine, with appropriate modifications to the file paths.

Install via:
```
remotes::install_github("matsengrp/sumrep@packagify-again")
```

However, this won't build the man pages.  Do a full install with roxygenization (man pages) as follows from the root of the cloned repo:
```
library(devtools)
install_deps(dep=T)
document()
build()
install()
```

Or just click _Build | Install & Restart_ in RStudio.

There are import warnings upon build, but you can ignore them for now. I'll fix them after we merge this in. Next steps are:

1. Update documentation.
2. Fix import warnings.
3. Generate a reference manual docs site.
4. Figure out what does/doesn't need to exported. I just exported everything with a roxygen block for now. I suspect that's overkill.
5. Ensure compatibility with lasts AIRR format compatible versions of alakazam and shazam.

I'll work on the above after we merge this in.

BrandenOlson commented:
So I keep getting errors similar to the following:

Error: Dependency package(s) 'pegas','plotmo','viridis' not available.

Not sure if this has to do with any of your changes, but I only get them on this branch and not on `master`.

javh commented:
Try this:

```
library(devtools)
install_deps(dep=T)
```

javh commented:
Some notes from working through the import conflicts:

* Where is the function `getCDR3s`?
* Consider replacing `multiplot` with `cowplot::plot_grid`, which is more powerful and would let you drop the `grid` dependency.
* Consider replacing `seqinr` functions with `Biostrings` equivalents (if practical).
* Dependencies `pegas`, `RecordLinkage`, and `viridis` don't appear to be used? At least that I can tell.

javh commented:
Finished the following:

- Updated `docs/usage.md` with new file loading commands.
- Fixed all the import warnings and dependency issues.
- Checked the alakazam and shazam calls. They seem fine.
- Added install of the sumrep package to the Dockerfile. (Not tested yet.)

I think that's enough for now. Test and merge maybe?

BrandenOlson commented:
Comments:

(1) It looks like `devtools` needs to be loaded for the following command:
``` remotes::install_github("matsengrp/sumrep@packagify-again") ```
Just a heads-up so we remember to make this clear on the installation page.
(2) I'm getting the following error when running the above command:
``` Error: Failed to install 'sumrep' from GitHub: 
  (converted from warning) package ‘Biostrings’ is not available (for R version 3.6.0) ```

javh commented:
> package ‘Biostrings’ is not available (for R version 3.6.0)

This is because Biostrings is in BioConductor, rather than CRAN proper. I pushed a change that should hopefully pull dependencies from bioconductor as well.

Though, if Biostrings is the only BioC dependency, then the ideal solution is to remove it and replace the functionality with something from another package. Is Biostrings used for anything other than the `getMotifCount` function?

BrandenOlson commented:
Ok, I'm getting a successful build now.

> Is Biostrings used for anything other than the getMotifCount function?

No, I'm pretty sure it's only used for `DNAStringSet` and `vcountPattern` in `getMotifCount`.

I'm getting some failed tests in `test.getColdspotCountDistribution` and `test.getHotspotCountDistribution` when running `devtools::test()`, so I'll figure those out and then run the examples in the quickstart. Did you want me to test the Docker build via the updated Dockerfile?

javh commented:
Okay, I'll look at `Biostrings::vcountPattern` and see if it's the same thing as `alakazam::countPatterns`.

I didn't check the tests. Oops.

And, yes on the docker build, but let me build first to make sure the install command I added worked. I'll do that right now and update as soon as it finishes.

BrandenOlson commented:
The hot and cold spot count tests required a change in the logic (i.e. I don't think your changes broke them) but I've pushed a fix.

The partis tests seem to be breaking because the `test_data.fa` file was removed.

javh commented:
> The partis tests seem to be breaking because the test_data.fa file was removed.

This is embedded in the package now under `inst/extdata`, the path to which can be retrieved in R via:

```
system.file("extdata/test_data.fa", package="sumrep")
```

If this needs to be accessible outside R (or a wrapper called from R), then maybe it should be copied into `Examples`?

javh commented:
BTW - Docker image isn't building for me. Think I've gotten past the firewall download issues and am getting actual installation errors now... Still working on it.

BrandenOlson commented:
Ok, I figured out the partis test issues and it looks fine to me. I'll wait to hear back about the Docker testing and we should be good to merge.

javh commented:
I think I have all the docker issues fixed, but I need to test a complete start-to-finish build. That'll take some time, so I'll update later.

javh commented:
Docker build now works for me.

BrandenOlson commented on Dockerfile:
```diff
@@ -95,21 +95,23 @@ WORKDIR /partis
-RUN git clone https://github.com/matsengrp/sumrep.git
-COPY . /sumrep
+# Download sumrep
+RUN git clone --single-branch --branch packagify-again https://github.com/matsengrp/sumrep.git /sumrep
```
Will need to update this when merged into master

BrandenOlson commented:
Okay @javh , I was able to build the Docker image and run a test script successfully (with a small tweak to the dockerfile).

This looks good to me. You have my 👍  to merge whenever you're ready. Lmk if I can help.

javh commented:
Great! I'll merge sometime before Thanksgiving and switch the branch in the docker file when I do.

javh commented:
Merged in. Running a test on the docker build from master and I'll push the changes once that passes.

Now that we've merged, I'll try to come back to the documentation tasks in a couple weeks (vignettes, man pages). I'll pester you then about what functions (don't) need to be exported and such.
